core/src/main/java/org/apache/spark/shuffle/api/ShuffleExecutorComponents.java

```diff
@@ -73,4 +77,30 @@ ShuffleMapOutputWriter createMapOutputWriter(
       long mapId) throws IOException {
     return Optional.empty();
   }
+
+  /**
+   * Returns an underlying {@link Iterable<java.io.InputStream>} that will iterate
+   * through shuffle data, given an iterable for the shuffle blocks to fetch.
+   */
+  <K, V, C> Iterable<ShuffleBlockInputStream> getPartitionReaders(
+      Iterable<ShuffleBlockInfo> blockInfos,
+      ShuffleDependency<K, V, C> dependency,
```

Comment:
Having this in the API doesn't seem ideal - but looking at the implementation of `LocalDiskShuffleExecutorComponents`, and how that inspects the `Serializer` to check for continuous block fetching.

Reply:
Just a thought.. Can passing `ShuffleHandle` make it look better?  
It looks like it is used to decide if batch Fetch Can Be supported Or Not.  The code at ShuffleBlockFetcherIterable as of this review, is more static(based on bunch of configs). Cant the decision be made early, then being done where ever its done now in this patch?

There could also be some "run time" difficulty in supporting batchFetch from the storage perspective, which i feel shuffleMetaData must answer, "Can you support batch fetch for these blocks now?" Sometimes may be yes, sometimes no... 

May be there are changes in further PRs, but just recording i for now.

Reply:
I don't _think_ option 2 is possible, because the serializer can be configured on each shuffle. The user technically has the ability now to set the serializer on the shuffle handle backing the shuffled RDD. I don't know how often this is done in practice, but since this is a public facing API, I think we need to not break such use cases. Does that make sense?

I don't have a strong opinion on `ShuffleHandle` vs. `ShuffleDependency`, is there a reason why it may be more elegant to pass along one or the other? They're both internal-ish APIs.